dotcom-rendering/src/components/LiveblogRightAdSlots.importable.tsx

```diff
@@ -0,0 +1,36 @@
+import { useAB } from '../lib/useAB';
+import { AdSlot } from './AdSlot';
+import { LiveblogRightMultipleAdSlots } from './LiveblogRightMultipleAdSlots';
+
+type Props = {
+	display?: ArticleDisplay;
+	isPaidContent?: boolean;
+};
+
+export const LiveblogRightAdSlots = ({ display, isPaidContent }: Props) => {
```

Comment:
This is just a suggestion, but I wonder if it'd be clearer if all of the AB test variant checking happened in one place? So there's 3 ways of rendering right-hand ads on liveblogs:

```typescript
export const LiveblogRightAdSlots = ({ display, isPaidContent }: Props) => {
	const ABTestAPI = useAB()?.api;

	const shouldInsertMultipleAdverts =
		ABTestAPI?.isUserInVariant(
			'LiveblogRightColumnAds',
			'multiple-adverts',
		) ?? false;

	if (shouldInsertMultipleAdverts) {
			return (
			  <LiveblogRightMultipleAdSlots
				  display={display}
				  isPaidContent={isPaidContent}
			  />
		);
	}

	const restrictStickyHeight =
		ABTestAPI?.isUserInVariant(
			'LiveblogRightColumnAds',
			'minimum-stickiness',
		) ?? false;

    // Bypass the <AdSlot> component completely
    // Since we know this must be a right hand column on a liveblog
	return (
		<TopRightAdSlot
			isPaidContent={isPaidContent}
			adStyles={[labelStyles]}
			restrictStickyHeight={restrictStickyHeight}
		/>
	);
};
```

This would require adding a new prop `restrictStickyHeight` to `<TopRightAdSlot>`, but I think you could get away without wrapping it in an Island then as well.

Reply:
Yeah I like this 👍